src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -117,5 +120,152 @@ private static ImmutableArray<ParameterName> GenerateNames(IList<string> reserve
 
             return GenerateNames(reservedNames, isFixed, parameterNames, parameterNamingRule);
         }
+
+        /// <summary>
+        /// Given an argument node, tries to generate an appropriate name that can be used for that
+        /// argument.
+        /// </summary>
+        public static string GenerateNameForArgument(
```

Comment:
@CyrusNajmabadi all code moved as is from master snapshot

Reply:
<3